unittest/CMakeLists.txt

```diff
@@ -10,38 +10,67 @@
 # or implied. See the License for the specific language governing permissions and limitations under the License
 
 include_directories(${KNOWHERE_SOURCE_DIR}/thirdparty)
-include_directories(${KNOWHERE_SOURCE_DIR}/thirdparty/SPTAG/AnnService)
-include_directories(${KNOWHERE_SOURCE_DIR}/thirdparty/NGT/lib)
+if (NOT APPLE)
+    include_directories(${KNOWHERE_SOURCE_DIR}/thirdparty/SPTAG/AnnService)
+    include_directories(${KNOWHERE_SOURCE_DIR}/thirdparty/NGT/lib)
+endif()
 include_directories(${KNOWHERE_SOURCE_DIR}/knowhere)
 include_directories(${KNOWHERE_SOURCE_DIR})
 
-set(DEPEND_LIBS
-        gtest
-        gmock
-        gtest_main
-        gmock_main
-        faiss
-        ngt
-        knowhere_log
-        openblas
-        )
+if (APPLE)
+    set(DEPEND_LIBS
+            gtest
+            gmock
+            gtest_main
+            gmock_main
+            knowhere_log
+            )
+else()
+    set(DEPEND_LIBS
+            gtest
+            gmock
+            gtest_main
+            gmock_main
+            faiss
+            ngt
+            knowhere_log
+            openblas
+            )
+endif()
 
 set(BASIC_LIBS gomp gfortran pthread)
 
-set(UTIL_SRCS
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/index/vector_index/adapter/VectorAdapter.cpp
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/index/vector_index/helpers/FaissIO.cpp
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/index/vector_index/helpers/IndexParameter.cpp
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/index/vector_index/helpers/DynamicResultSet.cpp
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/index/vector_index/Statistics.cpp
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/index/IndexType.cpp
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/common/Exception.cpp
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/common/Timer.cpp
-        ${KNOWHERE_SOURCE_DIR}/knowhere/knowhere/common/Utils.cpp
-        ${KNOWHERE_SOURCE_DIR}/unittest/utils.cpp
-        )
+if (APPLE)
```

Comment:
set ( UTIL_SRCS
    xxx      // ==> shared files for MAC and LINUX
    xxx
)
if ( LINUX )
    set ( UTIL_SRCS
        ${UTIL_SRCS}
        xxx    // files needed by LINUX only
        xxx
    )
endif ()

Reply:
Done.